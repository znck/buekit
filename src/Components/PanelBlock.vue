<script>
import PropTypes from '@znck/prop-types'
import { isString } from 'lodash-es'
import { style, styleResolver } from '../utils'
import { createTag } from '../mixins'

export default {
  name: 'PanelBlock',
  functional: true,
  mixins: [createTag('div', true)],
  props: {
    icon: PropTypes.bool.value(false),
    tabs: PropTypes.bool.value(false)
  },
  render (h, ctx) {
    const { icon, tabs } = ctx.props
    const s = styleResolver(ctx.$style)
    const styles = [
      icon && s('panel-icon'),
      tabs && s('panel-tabs'),
      !icon && !tabs && s('panel-block')
    ]

    if (isString(ctx.props.tag)) {
      return style(
        h(icon ? 'span' : ctx.props.tag, ctx.data, ctx.children),
        styles
      )
    }

    if (process.env.NODE_ENV !== 'production' && (!ctx.children || ctx.children.length !== 1)) {
      console.warn('Title is an abstract component. It requires exactly one child.')

      return null
    }

    return style(ctx.children[0], styles)
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/components/panel.sass';
</style>
