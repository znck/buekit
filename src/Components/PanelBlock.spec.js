import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'

import PanelBlock from './PanelBlock.vue'

describe('PanelBlock.vue', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(PanelBlock, {})
		const renderer = createRenderer()

		renderer.renderToString(wrapper.vm, (err, str) => {
			if (err) {
				throw new Error(err)
			}
			expect(str).toMatchSnapshot()
		})
	})
})
