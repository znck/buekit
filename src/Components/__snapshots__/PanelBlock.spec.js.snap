// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelBlock.vue matches snapshot 1`] = `<div class="panel-block"></div>`;
